### ISSUE TITLE
Underflow with torch.rand()?

### ISSUE THREAD

**Issue by zhuchen03** [pytorch/pytorch#2604]
Hi all, I am trying to use torch.rand() to implement Gumbel softmax, when I encountered a runtime error traced back to torch.rand(). It should generate random numbers from [0,1), but sometimes a very large number (e.g., 1069886848.0) occurs and my program crashes. 

Here is how I used it:

```python
def sample_gumbel(shape, eps=1e-10):
    U = torch.rand(*shape) * (1-eps)
    return -torch.log(-torch.log(U + eps) + eps)
```

I have also attached the problematic U. You can load it with pickle.
[U_problem.pkl.zip](https://github.com/pytorch/pytorch/files/1272683/U_problem.pkl.zip)

Any suggestions?

**Comment by apaszke** [comment]
I can't reproduce this. I've been running `torch.rand(1000, 1000).max()` in a loop and the result is never larger than 1.

How did you install PyTorch? What's `shape` in your example? What are your system specs and what Python version did you use?

**Comment by zhuchen03** [comment]
I installed by downloading http://download.pytorch.org/whl/cu80/torch-0.2.0.post3-cp27-cp27mu-manylinux1_x86_64.whl and install. 

shape is (24,81).

system:Linux  2.6.32_1-16-0-0
Python version: Python 2.7.13 |Anaconda 4.3.1 (64-bit)| (default, Dec 20 2016, 23:09:15) 

The error is quite random. Sometimes it occurs in tens of iterations, sometimes hundreds.

**Comment by yf225** [comment]
@zhuchen03 Curious is this problem still reproducible on PyTorch 1.4?

**Comment by zhuchen03** [comment]
Thanks a ton for following up! That's a long time ago. I do not have access to the environment any more, but I've never had the same issue again.
